src/firecracker/examples/uffd/uffd_utils.rs

```diff
@@ -263,13 +330,121 @@ impl UffdHandler {
     }
 }
 
+struct UffdMsgIterator {
+    stream: UnixStream,
+    buffer: Vec<u8>,
+    fds_received: bool,
+    secret_free: bool,
+    current_pos: usize,
+    fds: [RawFd; 3],
+}
+
+impl Iterator for UffdMsgIterator {
+    type Item = UffdMsgFromFirecracker;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if !self.fds_received {
+            let mut iov = [];
+            let mut fds = [0; 3];
+            let ret = unsafe { self.stream.recv_with_fds(&mut iov, &mut fds) };
+            match ret {
+                Ok((0, 1)) => {
+                    self.secret_free = false;
+                }
+                Ok((0, 3)) => {
+                    self.secret_free = true;
+                }
+                Ok((n, _)) => {
```

Comment:
`n` is the number of bytes read, so this should be `(_, n)` instead

Reply:
done, thanks.